src/stactools/nisar_sim/stac.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import re
+
+from pystac import Collection, Item, Summaries
+from pystac.extensions.item_assets import ItemAssetsExtension
+from pystac.extensions.projection import ProjectionExtension
+from pystac.extensions.sar import SarExtension
+from pystac.extensions.sat import SatExtension
+
+from stactools.nisar_sim import constants as c
+from stactools.nisar_sim.metadata import (
+    AnnotatedMetadata,
+    HDF5Metadata,
+    Metadata,
+    MetadataLinks,
+    fill_sar_properties,
+    fill_sat_properties,
+)
+from stactools.nisar_sim.nisar_assets import get_assets
+
+# import rasterio
```

Comment:
Remove if not using. 

Open question for later - can we get what we need without h5py? (I know gdalinfo can extract some info from hdf5 files)